backend/inspirehep/disambiguation/tasks.py

```diff
@@ -90,6 +117,58 @@ def disambiguate_authors(self, record_uuid):
             author_control_number = matched_authors[0]
             author["record"] = get_ref_from_pid("aut", author_control_number)
             updated_authors.append(str(author_control_number))
+            continue
```

Comment:
we should separate the logic in smaller functions, it became too long.

Reply:
I fully agree. If you look at what this function does from a high-level perspective, it:
*  does some initialization
* loops on all authors in the record
* inside he loop, it does a series of matches, if there's only one result, it assigns it, if not it goes to the next one, if no match, creates a new author
* does some final stuff

You could 
1. split the loop body into a `disambiguate_literature_author` which takes an element of `authors` and mutates it to add the ref or do nothing, and also return the ref/recid of the matched author (so you could still do the `updated_authors` logic).
2. refactor this `disambiguate_literature_author` to make the logic more apparent. There's lots of repetition because you call `match_literature_author` with different configs. I'd change this to take a `list` of configs, and return the first match (look at the reference matcher, it's a very similar idea).
3. split the author record creation in a separate function.